src/wazuh.py

```diff
@@ -283,91 +343,126 @@ def _get_current_configuration_url_branch(container: ops.Container) -> str:
         container.push(
-            RSA_PATH,
-            f"{custom_config_ssh_key}\n",
+            KNOWN_HOSTS_PATH,
+            output,
             encoding="utf-8",
             make_dirs=True,
             user=WAZUH_USER,
             group=WAZUH_GROUP,
             permissions=0o600,
         )
-
-    base_url = None
-    branch = None
-    try:
-        if custom_config_repository:
-            url = urlsplit(custom_config_repository)
-            path_parts = url.path.split("@")
-            branch = path_parts[1] if len(path_parts) > 1 else None
-            base_url = urlunsplit(url._replace(path=path_parts[0]))
-            process = container.exec(["ssh-keyscan", "-t", "rsa", str(url.hostname)], timeout=10)
-            output, _ = process.wait_output()
-            container.push(
-                KNOWN_HOSTS_PATH,
-                output,
-                encoding="utf-8",
-                make_dirs=True,
-                user=WAZUH_USER,
-                group=WAZUH_GROUP,
-                permissions=0o600,
-            )
-        if (
-            _get_current_configuration_url(container) != base_url
-            or _get_current_configuration_url_branch(container) != branch
-        ):
-            process = container.exec(["rm", "-Rf", REPOSITORY_PATH], timeout=1)
-            process.wait_output()
-
-            if base_url:
-                command = ["git", "clone", "--depth", "1"]
-                if branch:
-                    command = command + ["--branch", branch]
-                command = command + [base_url, REPOSITORY_PATH]
-                process = container.exec(command, timeout=60)
-                process.wait_output()
+        container.exec(["rm", "-Rf", REPOSITORY_PATH], timeout=1).wait_output()
+        command = ["git", "clone", "--depth", "1"]
+        if ref is not None:
+            command = command + ["--branch", ref]
+        command = command + [base_url, REPOSITORY_PATH]
+        container.exec(command, timeout=60).wait_output()
     except ops.pebble.ExecError as ex:
+        logger.error("git clone of custom_config_repository failed")
         raise WazuhInstallationError from ex
 
 
-def pull_configuration_files(container: ops.Container) -> None:
-    """Pull configuration files from the repository.
+def sync_config_repo(
+    container: ops.Container,
+    repository: typing.Optional[AnyUrl],
+    repo_ssh_key: typing.Optional[str],
+) -> bool:
+    """Synchronize a local copy of the custom config repository.
+
+    Args:
+        container: the container to configure git for.
+        repository: the git repository to add to known hosts in format
+        git+ssh://<user>@<host>@<branch|tag>.
+        repo_ssh_key: the SSH key for the git repository.
+
+    Returns:
+        bool: True if a pull was executed, False if there was nothing to sync
+
+    Raises:
+        WazuhInstallationError: if an error occurs while configuring git.
+    """
+    if repository is None:
```

Comment:
```suggestion
    if not repository:
```
we might want to catch empty strings also here